src/edit_diff.c

```diff
@@ -979,7 +979,7 @@ sr_edit_userord_is_moved(const struct lyd_node *data_match, enum insert_val inse
     case INSERT_FIRST:
     case INSERT_AFTER:
         sibling = sr_edit_find_previous_instance(data_match);
-        if (sibling == anchor_node) {
+        if (lyd_compare_single(sibling, anchor_node, LYD_PRINT_WD_ALL) == LY_SUCCESS) {
```

Comment:
What about the flag `LYD_PRINT_WD_ALL`? It is wrong to use here (notice it starts `LYD_PRINT...` and not `LYD_COMPARE...`) and does not map to any relevant flag anyway, so just use `0`. While at it, use just `!lyd_compare_single()` instead of the comparison, redundant.